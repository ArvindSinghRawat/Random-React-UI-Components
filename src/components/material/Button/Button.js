import React from 'react';
import PropTypes from 'prop-types';

// Styles
import styles from './Button.module.css';

const colorCss = {
  primary: styles.Primary,
  dark: styles.Dark,
  success: styles.Success,
  warning: styles.Warning,
  danger: styles.Danger
};

const typesCss = {
  outline: styles.Outline,
  none: styles.None
};

export const possibleTypes = Object.keys(typesCss);
export const possibleColor = Object.keys(colorCss);

export const Button = (props) => {
  const cssClasses = [styles.Button];

  const type = [];
  const color = [];

  const receivedType = props.type ? props.type.split(/[ ,-]+/) : [];
  for (const element of receivedType) {
    let temp = possibleColor.indexOf(element);
    if (temp !== -1) {
      color.push(colorCss[element]);
    } else {
      temp = possibleTypes.indexOf(element);
      if (temp !== -1) {
        type.push(typesCss[element]);
      } else {
        console.log(`${element} not known as type`);
      }
    }
  }

  if (color.length > 1) {
    console.log(
      `Should receive only a single Color class but received ${color.length}. But Classes Received: ${color}`
    );
    cssClasses.push(color.pop());
  } else if (color.length === 1) {
    cssClasses.push(color.pop());
  }

  if (type.length > 1) {
    console.log(
      `Should receive only a single Type class but received ${type.length}. But Classes Received: ${type}`
    );
    cssClasses.push(type.pop());
  } else if (type.length === 1) {
    cssClasses.push(type.pop());
  }

  const finalClass = (
    cssClasses.join(' ') + (props.className ? ' ' + props.className : '')
  ).trimRight();

  return (
    <button
      className={finalClass}
      style={{ ...props.style }}
      onClick={props.onClick ? props.onClick : (event) => {}}
      {...props.htmlAttributes}
    >
      {props.children}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object,
  htmlAttributes: PropTypes.object,
  onClick: PropTypes.func,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node])
};

export default Button;
