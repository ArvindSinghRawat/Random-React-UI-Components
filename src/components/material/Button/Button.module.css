/* Default or Primary Styling */

.Button {
  background-color: #007eff;
  color: #fff;
  border-radius: 5px;
  box-shadow: 0 5px 5px #99cbff;
  appearance: none;
  outline: none;
  border: none;
  transition: all 0.1s ease-out;
}

.Button:active {
  box-shadow: 0 2px 1px #99cbff, inset 0 3px 2px #006dee;
  transform: translateY(3px);
}

.Button.Outline {
  background-color: #fff;
  border: 2px solid #007eff;
  color: #007eff;
}

.Button.Outline:active {
  box-shadow: 0 2px 2px #99cbff, inset 0 3px 4px #99cbff;
  transform: translateY(3px);
}

.Button.None {
  background-color: transparent;
  border: none;
  color: #007eff;
  box-shadow: none;
  transform: none;
}

.Button.None:hover {
  box-shadow: 0 1px 2px 1px #eee !important;
}

.Button.None:active {
  color: #008fff;
  box-shadow: none;
  transform: translateY(3px);
  box-shadow: inset 0 1px 2px 2px #eee !important;
}

/* Dark Button Styling */

.Button.Dark {
  background-color: #255763;
  color: #fff;
  box-shadow: 0 5px 3px #2e6d7c;
  transition: all 0.1s ease-out;
}

.Button.Dark:active {
  box-shadow: 0 2px 1px #2e6d7c, inset 0 3px 3px #1c414a;
  transform: translateY(3px);
}

.Button.Dark.Outline {
  background-color: #fff;
  border: 2px solid #255763;
  color: #255763;
  box-shadow: 0 5px 3px #a0a4a8;
}

.Button.Dark.Outline:active {
  box-shadow: 0 2px 2px #99cbff, inset 0 3px 4px #99cbff;
  transform: translateY(3px);
}

.Button.Dark.None {
  background-color: transparent;
  border: none;
  color: #255763;
  box-shadow: none;
  transform: none;
}

.Button.Dark.None:active {
  color: #255763;
  box-shadow: 0 2px 2px #99cbff, inset 0 3px 4px #99cbff;
  transform: translateY(3px);
}

/* Dark Button Styling */

.Button.Success {
  background-color: #4cbb17;
  color: #fff;
  box-shadow: 0 5px 4px #40dd7f;
  transition: all 0.1s ease-out;
}

.Button.Success:active {
  box-shadow: 0 2px 1px#40dd7f, inset 0 3px 3px #708238;
  transform: translateY(3px);
}

.Button.Success.Outline {
  background-color: #fff;
  border: 2px solid #4cbb17;
  color: #4cbb17;
  box-shadow: 0 5px 6px #40dd7f;
}

.Button.Success.Outline:active {
  box-shadow: 0 2px 2px #40dd7f, inset 0 3px 4px #40dd7f;
  transform: translateY(3px);
}

.Button.Success.None {
  background-color: transparent;
  border: none;
  color: #4cbb17;
  box-shadow: none;
  transform: none;
}

.Button.Success.None:hover {
  box-shadow: 0 2px 2px #40dd7f !important;
}

.Button.Success.None:active {
  color: #4cbb17;
  box-shadow: 0 1px 2px #40dd7f, inset 0 1px 2px #40dd7f !important;
  transform: translateY(3px);
}

/* Warning Button Styling */

.Button.Warning {
  background-color: #f6a609;
  color: #fff;
  box-shadow: 0 5px 4px #e89806;
  transition: all 0.1s ease-out;
}

.Button.Warning:active {
  box-shadow: 0 2px 1px#e89806, inset 0 3px 2px #e89806;
  transform: translateY(3px);
}

.Button.Warning.Outline {
  background-color: #fff;
  border: 2px solid #f6a609;
  color: #f6a609;
  box-shadow: 0 5px 6px #ffbc1f;
}

.Button.Warning.Outline:active {
  box-shadow: 0 2px 2px #ffbc1f, inset 0 2px 2px #ffbc1f;
  transform: translateY(3px);
}

.Button.Warning.None {
  background-color: transparent;
  border: none;
  color: #f6a609;
  box-shadow: none;
  transform: none;
}

.Button.Warning.None:hover {
  box-shadow: 0 2px 2px #ffbc1f !important;
}

.Button.Warning.None:active {
  color: #4cbb17;
  box-shadow: 0 1px 2px #ffbc1f, inset 0 1px 2px #ffbc1f !important;
  transform: translateY(3px);
}

/* Danger Button Styling */

.Button.Danger {
  background-color: #ff2400;
  color: #fff;
  box-shadow: 0 5px 5px #ee2424;
  transition: all 0.1s ease-out;
}

.Button.Danger:active {
  box-shadow: 0 2px 2px#ee2424, inset 0 3px 2px #e93c3c;
  transform: translateY(3px);
}

.Button.Danger.Outline {
  background-color: #fff;
  border: 2px solid #ff2400;
  color: #ff2400;
  box-shadow: 0 4px 5px #ff6262;
}

.Button.Danger.Outline:active {
  box-shadow: 0 2px 2px #ff6262, inset 0 3px 3px #ff6262;
  transform: translateY(3px);
}

.Button.Danger.None {
  background-color: transparent;
  border: none;
  color: #ff2400;
  box-shadow: none;
  transform: none;
}

.Button.Danger.None:hover {
  box-shadow: 0 2px 2px #ff6262 !important;
}

.Button.Danger.None:active {
  color: #fb4e4e;
  box-shadow: 0 1px 2px #ff6262, inset 0 1px 2px #ff6262 !important;
  transform: translateY(3px);
}
