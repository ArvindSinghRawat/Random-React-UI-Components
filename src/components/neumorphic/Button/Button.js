import React from 'react';
import { buttonWrapper } from '../../../hoc/ButtonWrapper';

// Style
import styles from './Button.module.css';

const colorCss = {
  light: styles.Light,
  dark: styles.Dark,
  xdark: styles.XDark
};

const typesCss = {
  inset: styles.Inset,
  defaultIn: styles.DefaultIn,
  defaultOut: styles.DefaultOut
};

export const possibleTypes = Object.keys(typesCss);
export const possibleColor = Object.keys(colorCss);

const ButtonComponent = (props) => {
  console.log(props.finalClass);
  return (
    <button
      className={props.finalClass}
      style={{ ...props.style }}
      onClick={props.onClick ? props.onClick : (event) => {}}
      {...props.htmlAttributes}
    >
      {props.children}
    </button>
  );
};

ButtonComponent.displayName = 'Neumorphic Button';

export const Button = buttonWrapper(
  ButtonComponent,
  [styles.Button],
  colorCss,
  typesCss
);

export default Button;
