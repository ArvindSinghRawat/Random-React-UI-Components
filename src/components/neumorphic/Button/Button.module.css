/* Default White Styling */

.Button {
  color: #000;
  background-color: #fff;
  box-shadow: 4px 4px 5px 0 #d9d9d9, -4px -4px 5px 0 #fafafa;
  border-radius: 10px;
  border: none;
  outline: none;
  appearance: none;
  transition: all 0.1s ease-out;
}

.Button:active {
  box-shadow: 1px 1px 5px 0 #d9d9d9, -1px -1px 5px 0 #fafafa;
  transform: translateY(3px);
}

.Button.Inset {
  background-color: '#fff';
  box-shadow: inset 4px 4px 5px 0 #d9d9d9, inset -4px -4px 5px 0 #fafafa;
}

.Button.Inset:active {
  box-shadow: inset 7px 7px 8px 0 #d9d9d9, inset -7px -7px 8px 0 #fafafa;
  transform: translateY(3px);
}

.Button.DefaultIn:active {
  animation: defInToOut 0.1s linear;
  box-shadow: inset 4px 4px 5px 0 #d9d9d9, inset -4px -4px 5px 0 #fafafa;
}

@keyframes inToOut {
  0% {
    box-shadow: 4px 4px 5px 0 #d9d9d9, -4px -4px 5px 0 #fafafa;
  }
  25% {
    box-shadow: 2px 2px 2.5px 0 #d9d9d9, -2px -2px 2.5px 0 #fafafa;
  }
  50% {
    box-shadow: none;
  }
  75% {
    box-shadow: inset 2px 2px 2.5px 0 #d9d9d9, inset -2px -2px 2.5px 0 #fafafa;
  }
  100% {
    box-shadow: inset 4px 4px 5px 0 #d9d9d9, inset -4px -4px 5px 0 #fafafa;
  }
}

/* Dark Styling */

.Button.Dark {
  color: #fff;
  background-color: #255763;
  box-shadow: 4px 4px 5px 0 #2e6d7c, -4px -4px 5px 0 #1c414a;
}

.Button.Dark:active {
  box-shadow: 1px 1px 5px 0 #2e6d7c, -1px -1px 5px 0 #1c414a;
}

.Button.Dark.Inset {
  color: #fff;
  background-color: #255763;
  box-shadow: inset 4px 4px 5px 0 #1c414a, inset -4px -4px 5px 0 #2e6d7c;
}

.Button.Dark.Inset:active {
  box-shadow: inset 7px 7px 8px 0 #1c414a, inset -7px -7px 10px 0 #2e6d7c;
}

.Button.Dark.DefaultIn {
  box-shadow: 4px 4px 5px 0 #1c414a, -4px -4px 5px 0 #2e6d7c;
}

.Button.Dark.DefaultIn:active {
  animation: darkInToOut 0.1s linear;
  box-shadow: inset 7px 7px 8px 0 #1c414a, inset -7px -7px 10px 0 #2e6d7c;
}

@keyframes darkInToOut {
  0% {
    box-shadow: 4px 4px 5px 0 #1c414a, -4px -4px 5px 0 #2e6d7c;
  }
  25% {
    box-shadow: 2px 2px 2.5px 0 #1c414a, -2px -2px 2.5px 0 #2e6d7c;
  }
  50% {
    box-shadow: none;
  }
  75% {
    box-shadow: inset 1px 1px 2.5px 0 #1c414a, inset -1px -1px 2.5px 0 #2e6d7c;
  }
  100% {
    box-shadow: inset 1px 1px 5px 0 #1c414a, inset -1px -1px 5px 0 #2e6d7c;
  }
}

/* Extra Dark Styling */

.Button.XDark {
  color: #fff;
  background-color: #0a0a0a;
  box-shadow: 4px 4px 5px 0 rgba(255, 255, 255, 0.025),
    -4px -4px 5px 0 rgba(0, 0, 0, 0.5);
}

.Button.XDark:active {
  box-shadow: 1px 1px 5px 0 rgba(255, 255, 255, 0.025),
    -1px -1px 5px 0 rgba(0, 0, 0, 0.5);
}

.Button.XDark.Inset {
  color: #fff;
  background-color: #0a0a0a;
  box-shadow: inset 4px 4px 4px 0 rgba(0, 0, 0, 0.001),
    inset -4px -4px 5px 0 rgba(0, 0, 0, 0.5);
}

.Button.XDark.Inset:active {
  box-shadow: inset 7px 7px 8px 0 rgba(0, 0, 0, 0.33),
    inset -7px -7px 8px 0 rgba(0, 0, 0, 0.66);
}

.Button.XDark.DefaultIn {
  box-shadow: 4px 4px 5px 0 rgba(255, 255, 255, 0.025),
    -4px -4px 5px 0 rgba(0, 0, 0, 0.5);
}

.Button.XDark.DefaultIn:active {
  animation: xDarkInToOut 0.1s linear;
  box-shadow: inset 7px 7px 8px 0 rgba(0, 0, 0, 0.33),
    inset -7px -7px 8px 0 rgba(0, 0, 0, 0.5);
}

@keyframes xDarkInToOut {
  0% {
    box-shadow: 4px 4px 5px 0 rgba(255, 255, 255, 0.025),
      -4px -4px 5px 0 rgba(0, 0, 0, 0.5);
  }
  25% {
    box-shadow: 4px 4px 5px 0 rgba(255, 255, 255, 0.015),
      -4px -4px 5px 0 rgba(0, 0, 0, 0.54);
  }
  50% {
    box-shadow: 4px 4px 5px 0 rgba(255, 255, 255, 0.005),
      -4px -4px 5px 0 rgba(0, 0, 0, 0.58);
  }
  75% {
    box-shadow: inset 7px 7px 8px 0 rgba(0, 0, 0, 0.01),
      inset -7px -7px 8px 0 rgba(0, 0, 0, 0.62);
  }
  100% {
    box-shadow: inset 7px 7px 8px 0 rgba(0, 0, 0, 0.33),
      inset -7px -7px 8px 0 rgba(0, 0, 0, 0.66);
  }
}

/* Light Styling */

.Button.Light {
  color: #000;
  background-color: #e5e0ec;
  box-shadow: 4px 4px 5px 0 #a3b1e6, -4px -4px 5px 0 rgba(250, 250, 250, 0.3);
  transition: all 0.1s linear;
}

.Button.Light:active {
  box-shadow: 1px 1px 5px 0 #a3b1e6, -1px -1px 5px 0 rgba(250, 250, 250, 0.3);
}

.Button.Light.Inset {
  box-shadow: inset 4px 4px 5px 0 #a3b1e6,
    inset -4px -4px 5px 0 rgba(250, 250, 250, 0.3);
}

.Button.Light.Inset:active {
  box-shadow: inset 7px 7px 8px 0 #a3b1e6,
    inset -7px -7px 8px 0 rgba(250, 250, 250, 0.5);
}

.Button.Light.DefaultIn {
  box-shadow: 4px 4px 5px 0 #a3b1e6, -4px -4px 5px 0 rgba(250, 250, 250, 0.3);
}

.Button.Light.DefaultIn:active {
  animation: lightInToOut 0.1s linear;
  box-shadow: inset 4px 4px 5px 0 #a3b1e6,
    inset -4px -4px 5px 0 rgba(250, 250, 250, 0.3);
}

@keyframes lightInToOut {
  0% {
    box-shadow: 4px 4px 5px 0 #a3b1e6, -4px -4px 5px 0 rgba(250, 250, 250, 0.3);
  }
  25% {
    box-shadow: 2px 2px 2.5px 0 #a3b1e6,
      -2px -2px 2.5px 0 rgba(250, 250, 250, 0.3);
  }
  50% {
    box-shadow: none;
  }
  75% {
    box-shadow: inset 1px 1px 2.5px 0 #a3b1e6,
      inset -1px -1px 2.5px 0 rgba(250, 250, 250, 0.3);
  }
  100% {
    box-shadow: inset 1px 1px 5px 0 #a3b1e6,
      inset -1px -1px 5px 0 rgba(250, 250, 250, 0.3);
  }
}
