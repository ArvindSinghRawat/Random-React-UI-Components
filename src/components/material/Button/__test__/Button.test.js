import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Button, possibleColor, possibleTypes } from '../Button';

configure({ adapter: new Adapter() });

describe('<Button /> : ', () => {
  let wrapper = null;
  const content = 'Click Me !';
  beforeEach(() => {
    wrapper = shallow(<Button>{content}</Button>);
  });

  it('Should correctly render <Button />', () => {
    expect(wrapper.text()).toMatch(content);
    expect(wrapper).toMatchSnapshot();
  });

  for (const color of possibleColor) {
    it(`Should correctly render <Button type="${color}"/> for each type`, () => {
      wrapper.setProps({ type: color });
      expect(wrapper).toMatchSnapshot();
    });

    for (const type of possibleTypes) {
      it(`Should correctly render <Button type="${color} ${type}"/> for each color-type combination`, () => {
        wrapper.setProps({ type: `${color} ${type}` });
        expect(wrapper).toMatchSnapshot();
      });
    }
  }

  for (const type of possibleTypes) {
    it(`Should correctly render <Button type"${type}"/> for each type`, () => {
      wrapper.setProps({ type: `${type}` });
      expect(wrapper).toMatchSnapshot();
    });
  }

  it('Should correctly render <Button type="outline none"/> with console.error for mulitple types', () => {
    console.log = jest.fn();
    wrapper.setProps({ type: 'outline none' });
    expect(wrapper.text()).toMatch(content);
    expect(console.log).toHaveBeenCalled();
  });

  it('Should correctly render <Button type="dark success"/> with console.error for mulitple colors', () => {
    console.log = jest.fn();
    wrapper.setProps({ type: 'dark success' });
    expect(wrapper.text()).toMatch(content);
    expect(console.log).toHaveBeenCalled();
  });

  it('Should correctly render <Button /> even when unknown type is given with log', () => {
    console.log = jest.fn();
    wrapper.setProps({ type: 'xsaldnf' });
    expect(wrapper.text()).toMatch(content);
    expect(console.log).toHaveBeenCalled();
  });

  it('Should correctly render <Button className="customClass" />', () => {
    wrapper.setProps({ className: 'customClass' });
    expect(wrapper.text()).toMatch(content);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should correctly render <Button onClick="custommethod" />', () => {
    const customMethod = jest.fn();
    wrapper.find('button').simulate('click');
    expect(customMethod).not.toHaveBeenCalled();
    wrapper.setProps({ onClick: customMethod });
    expect(wrapper.text()).toMatch(content);
    expect(wrapper).toMatchSnapshot();
    expect(customMethod).not.toHaveBeenCalled();
    wrapper.find('button').simulate('click');
    expect(customMethod).toHaveBeenCalled();
  });

  it('Should return error for incorrect prop type />', () => {
    console.error = jest.fn();
    wrapper.setProps({ onClick: 'asdfs' });
    expect(console.error).toHaveBeenCalled();
  });
});
