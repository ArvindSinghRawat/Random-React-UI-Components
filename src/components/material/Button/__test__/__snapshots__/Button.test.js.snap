// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`For Material Button :  Should correctly render <Button /> 1`] = `
<ButtonWrapper(Material Button)>
  <Material Button
    finalClass="Button"
  >
    <button
      className="Button"
      onClick={[Function]}
      style={Object {}}
    >
      This is a Material Button
    </button>
  </Material Button>
</ButtonWrapper(Material Button)>
`;

exports[`For Material Button :  Should correctly render <Button onClick="custommethod" /> 1`] = `
<ButtonWrapper(Material Button)
  onClick={[MockFunction]}
>
  <Material Button
    finalClass="Button"
    onClick={[MockFunction]}
  >
    <button
      className="Button"
      onClick={[MockFunction]}
      style={Object {}}
    >
      This is a Material Button
    </button>
  </Material Button>
</ButtonWrapper(Material Button)>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "customClass" 1`] = `
<Material Button
  className="customClass"
  finalClass="Button customClass"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "danger" 1`] = `
<Material Button
  finalClass="Button Danger"
  type="danger"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "dark" 1`] = `
<Material Button
  finalClass="Button Dark"
  type="dark"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "none danger" 1`] = `
<Material Button
  finalClass="Button None Danger"
  type="none danger"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "none dark" 1`] = `
<Material Button
  finalClass="Button None Dark"
  type="none dark"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "none primary" 1`] = `
<Material Button
  finalClass="Button None Primary"
  type="none primary"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "none success" 1`] = `
<Material Button
  finalClass="Button None Success"
  type="none success"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "none warning" 1`] = `
<Material Button
  finalClass="Button None Warning"
  type="none warning"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "none" 1`] = `
<Material Button
  finalClass="Button None"
  type="none"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "outline danger" 1`] = `
<Material Button
  finalClass="Button Outline Danger"
  type="outline danger"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "outline dark" 1`] = `
<Material Button
  finalClass="Button Outline Dark"
  type="outline dark"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "outline primary" 1`] = `
<Material Button
  finalClass="Button Outline Primary"
  type="outline primary"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "outline success" 1`] = `
<Material Button
  finalClass="Button Outline Success"
  type="outline success"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "outline warning" 1`] = `
<Material Button
  finalClass="Button Outline Warning"
  type="outline warning"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "outline" 1`] = `
<Material Button
  finalClass="Button Outline"
  type="outline"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "primary" 1`] = `
<Material Button
  finalClass="Button Primary"
  type="primary"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "success" 1`] = `
<Material Button
  finalClass="Button Success"
  type="success"
>
  This is a Material Button
</Material Button>
`;

exports[`For Material Button :  Should correctly render Button with cssClass = "warning" 1`] = `
<Material Button
  finalClass="Button Warning"
  type="warning"
>
  This is a Material Button
</Material Button>
`;
